### PR TITLE
openlineage: add missing `dag_state_change_process_pool_size` in `provider.yaml`.

### PR DESCRIPTION
**Pull request by JDarDagran**
In #39235 I missed for some reason adding `dag_state_change_process_pool_size` in the `provider.yaml` file. This small PR fixes this.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/openlineage/provider.yaml [hunk 1]
```diff
@@ -135,3 +135,11 @@ config:
         example: ~
         type: boolean
         version_added: ~
+      dag_state_change_process_pool_size:
+        description: |
+          Number of processes to use for processing DAG state changes.
```
Comment (RNHTTR) [inline airflow/providers/openlineage/provider.yaml]: It might be helpful to describe what component these processes are being executed on

Comment (JDarDagran) [reply airflow/providers/openlineage/provider.yaml]: Rephrased it.

**Review by RNHTTR — APPROVED** [review]

File: airflow/providers/openlineage/provider.yaml [hunk 2]
```diff
@@ -135,3 +135,11 @@ config:
...
         type: boolean
         version_added: ~
+      dag_state_change_process_pool_size:
+        description: |
+          Number of processes to utilize for processing DAG state changes
+          in an asynchronous manner within the scheduler process.
+        default: "1"
+        example: ~
+        type: integer
+        version_added: ~
```
Comment (eladkal) [inline airflow/providers/openlineage/provider.yaml]: We should specify what version of the provider it was added

Comment (JDarDagran) [reply airflow/providers/openlineage/provider.yaml]: I couldn't find it in docs but should I guess which version (probably patch) would this be released in?

Comment (kacpermuda) [reply airflow/providers/openlineage/provider.yaml]: I think in #39609 we assumed next release as minor 1.8.0

**Review by mobuchowski — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**